app/src/main/java/com/adam/aslfms/util/Util.java

```diff
@@ -180,9 +180,10 @@ public static void warningDialog(Context ctx, String msg) {
 	public static void scrobbleIfPossible(Context ctx, NetApp napp,
 			int numInCache) {
 		if (numInCache > 0) {
-			Intent i = new Intent(ScrobblingService.ACTION_JUSTSCROBBLE);
-			i.putExtra("netapp", napp.getIntentExtraValue());
-			ctx.startService(i);
+            Intent intent = new Intent(ctx, ScrobblingService.class);
+			intent.setAction(ScrobblingService.ACTION_JUSTSCROBBLE);
+			intent.putExtra("netapp", napp.getIntentExtraValue());
+			ctx.startService(intent);

```

Comment:
I'm guessing when bumping up the targetSDK from 12 to 22, something changed in how you can start up services. I was getting this crash:

```  java.lang.IllegalArgumentException: Service Intent must be explicit: Intent { act=com.adam.aslfms.service.justscrobble (has extras) }
            at android.app.ContextImpl.validateServiceIntent(ContextImpl.java:1680)
            at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1709)
            at android.app.ContextImpl.startService(ContextImpl.java:1693)
            at android.content.ContextWrapper.startService(ContextWrapper.java:516)
            at com.adam.aslfms.util.Util.scrobbleAllIfPossible(Util.java:196)
            at com.adam.aslfms.SettingsActivity.onPreferenceTreeClick(SettingsActivity.java:116)
            at android.preference.Preference.performClick(Preference.java:993)
            at android.preference.PreferenceScreen.onItemClick(PreferenceScreen.java:214)
            at android.widget.AdapterView.performItemClick(AdapterView.java:305)
            at android.widget.AbsListView.performItemClick(AbsListView.java:1146)
            at android.widget.AbsListView$PerformClick.run(AbsListView.java:3053)
            at android.widget.AbsListView$3.run(AbsListView.java:3860)
            at android.os.Handler.handleCallback(Handler.java:739)
            at android.os.Handler.dispatchMessage(Handler.java:95)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5254)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698) ```

wherever we called `startService(Intent intent)`. All the Service Intents get resolved by ScrobblingService, so I might as well make all the Intents explicit.

Please double-check me and make sure this is the right approach before we merge in this branch.